### PR TITLE
Update  layout scale bar item description and screenshots

### LINKED ISSUE #2625: [FEATURE][layouts] Use text renderer for drawing scalebar text
Original commit: https://github.com/qgis/QGIS/commit/9d2c4fca1986a144f4e1cc8ada8a4450acb47e59 by nyalldawson

Allows the full range of formatting options exposed through
text renderer - e.g. scalebar text with buffers, shadows,
background shapes, letter spacing, etc.

Say goodbye to unreadable scale bar text!

### LINKED ISSUE #4047: [FEATURE][layout] Add above/below segments placement option for scale bar labels
Original commit: https://github.com/qgis/QGIS/commit/e5deb5517b83424f8a5558a1627917b35e149ef9 by nirvn

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #4048: [FEATURE][layout] Add horizontal placement option for scale bar labels
Original commit: https://github.com/qgis/QGIS/commit/3a37c83da757eb3492b8f7d7a232fd18d08dc377 by nirvn

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #4048 fixes #4047 fixes #2625

### CODE REVIEW

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 1]
```diff
@@ -15,8 +15,9 @@ Use the |scaleBar| :guilabel:`Add Scale Bar` tool following :ref:`items creation
 instructions <create_layout_item>` to add a new scale bar item that you can
 later manipulate the same way as exposed in :ref:`interact_layout_item`.
 
-By default, a new scale bar item shows the scale of the most recently added map
-item. You can customize it thanks to the :guilabel:`Item Properties` panel.
+By default, a new scale bar item shows the scale of the map item over which
+it's drawn. If no map item is below, the :ref:`reference map <reference_map>` is
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
it is drawn. If there is no map item below, the :ref:`reference map <reference_map>` is
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 2]
```diff
@@ -15,8 +15,9 @@ Use the |scaleBar| :guilabel:`Add Scale Bar` tool following :ref:`items creation
...
+used. You can customize it thanks to the :guilabel:`Item Properties` panel.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
used. You can customize it in the :guilabel:`Item Properties` panel.
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 3]
```diff
@@ -41,20 +42,21 @@ The :guilabel:`Main properties` group of the scale bar
 
    Scale Bar Main properties group
 
-* First, choose the map the scale bar will be attached to.
-* Then, choose the style of the scale bar. Six styles are available:
+#. First, choose the map the scale bar will be attached to.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
#. First, choose the map the scale bar will be attached to
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 4]
```diff
@@ -41,20 +42,21 @@ The :guilabel:`Main properties` group of the scale bar
...
+#. Then, choose the style of the scale bar. Six styles are available:
 
-  * **Single box** and **Double box** styles, which contain one or two lines of
-    boxes alternating colors;
-  * **Middle**, **Up** or **Down** line ticks;
-  * **Numeric**, where the scale ratio is printed (i.e., 1:50000).
+   * **Single box** and **Double box** styles, which contain one or two lines
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
   * **Single box** and **Double box** styles, which contain one (single)
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 5]
```diff
@@ -41,20 +42,21 @@ The :guilabel:`Main properties` group of the scale bar
...
+     of boxes alternating colors;
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
     or two (double) lines of boxes with alternating colors;
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 6]
```diff
@@ -41,20 +42,21 @@ The :guilabel:`Main properties` group of the scale bar
...
+#. Set properties as appropriate
 
 Units and Segments
 ------------------
 
 The :guilabel:`Units` and :guilabel:`Segments` groups of the scale bar
-:guilabel:`Item Properties` panel provide the following functionalities
-(see figure_layout_scalebar_units_):
+:guilabel:`Item Properties` panel (not available for the **Numeric** style)
+provide the following functionalities (see figure_layout_scalebar_units_):
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
provides the following functionalities (see figure_layout_scalebar_units_):
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 7]
```diff
@@ -100,12 +102,16 @@ You can define how the scale bar will be displayed in its frame.
 
 * :guilabel:`Box margin` : space between text and frame borders
 * :guilabel:`Labels margin` : space between text and scale bar drawing
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]:
```suggestion
* :guilabel:`Label margin` : space between text and scale bar drawing
```

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst [hunk 8]
```diff
@@ -122,11 +128,11 @@ The :guilabel:`Fonts and colors` group of the scale bar
...
    Scale Bar Fonts and colors groups
 
-You can define the fonts and :ref:`colors <color-selector>` used for the scale
-bar.
+You can define the :ref:`fonts <font_selector>` and :ref:`colors <color-selector>`
+used for the scale bar.
 
-* Use the :guilabel:`Font` button to set the font of scale bar label
-* :guilabel:`Font color`: set the font color
+* Use the :guilabel:`Font` button to set the :ref:`properties <text_format>`
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: ``Font`` looks like a pull-down menu, but acts like both a button and a pull-down menu - strange..

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst (hunk 4 above)
Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: looks redundant to me.

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst (hunk 6 above)
Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: we have two groups here, hence the pural

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst (hunk 8 above)
Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: like the color selector, the color ramp selector

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: In my opinion, it makes thinks slightly clearer. But no big deal.

File: source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/print_composer/composer_items/composer_scale_bar.rst]: QT I guess.

**Pull request merged**